reviewing/dependencies-hierarchy/src/index.ts

```diff
@@ -236,26 +224,47 @@ interface GetTreeOpts {
   wantedPackages: PackageSnapshots
 }
 
-interface DependencyInfo { circular?: true, dependencies: PackageNode[] }
+interface DependencyInfo {
+  dependencies: PackageNode[]
+
+  circular?: true
+
+  /**
+   * Whether or not the dependencies array was fully enumerated. This may not be
+   * the case if a max depth was hit.
+   */
+  isPartiallyVisited: boolean
+
+  /**
+   * The number of edges along longest path. null if the dependencies array is
+   * empty.
+   */
+  height: number | null
+}
 
 function getTree (
   opts: GetTreeOpts,
   keypath: string[],
   parentId: string
 ): PackageNode[] {
-  const dependenciesCache = new Map<string, PackageNode[]>()
+  const dependenciesCache = new DependenciesCache()
 
   return getTreeHelper(dependenciesCache, opts, keypath, parentId).dependencies
 }
 
 function getTreeHelper (
-  dependenciesCache: Map<string, PackageNode[]>,
+  dependenciesCache: DependenciesCache,
   opts: GetTreeOpts,
   keypath: string[],
   parentId: string
 ): DependencyInfo {
-  const result: DependencyInfo = { dependencies: [] }
-  if (opts.currentDepth > opts.maxDepth || !opts.currentPackages || !opts.currentPackages[parentId]) return result
+  if (opts.currentDepth > opts.maxDepth) {
+    return { dependencies: [], isPartiallyVisited: true, height: null }
+  }
+
+  if (!opts.currentPackages?.[parentId]) {
```

Comment:
After splitting the `opts.currentDepth > opts.maxDepth` condition out of the original `if` statement conditional, an ESLint rule recommended transforming:

```ts
!opts.currentPackages || !opts.currentPackages[parentId]
```

into:

```ts
!opts.currentPackages?.[parentId]
```